src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfEhFrame.cs

```diff
@@ -0,0 +1,203 @@
+                AddCie(fde.Cie);
+                cieOffset = _cieOffset[fde.Cie];
+            }
+
+            WriteFde(fde, cieOffset);
+        }
+
+        private void WriteCie(DwarfCie cie)
+        {
+            Utf8StringBuilder augmentationString = new Utf8StringBuilder();
+            uint augmentationLength = 0;
+            Span<byte> tempBuffer = stackalloc byte[8];
+
+            if (cie.FdesHaveAugmentationData)
+            {
+                augmentationString.Append('z');
+            }
+            if (cie.PersonalitySymbolName != null)
+            {
+                augmentationString.Append('P');
+                augmentationLength += 1u + AddressSize(cie.PersonalityEncoding);
+            }
+            if (cie.LsdaEncoding != 0)
+            {
+                augmentationString.Append('L');
+                augmentationLength++;
+            }
+            if (cie.PointerEncoding != 0)
+            {
+                augmentationString.Append('R');
+                augmentationLength++;
+            }
+            if (cie.IsSignalFrame)
+            {
+                augmentationString.Append('S');
+            }
+
+            uint length =
+                4u + // Length
+                4u + // CIE Offset (0)
+                1u + // Version
+                (uint)augmentationString.Length + 1u +
+                DwarfHelper.SizeOfULEB128(cie.CodeAlignFactor) +
+                DwarfHelper.SizeOfSLEB128(cie.DataAlignFactor) +
+                DwarfHelper.SizeOfULEB128(cie.ReturnAddressRegister) +
+                (uint)(augmentationLength > 0 ? DwarfHelper.SizeOfULEB128(augmentationLength) + augmentationLength : 0) +
+                (uint)cie.Instructions.Length;
+            uint padding = ((length + 7u) & ~7u) - length;
+
+            BinaryPrimitives.WriteUInt32LittleEndian(tempBuffer, length + padding - 4u);
+            _sectionWriter.Write(tempBuffer);
+
+            _sectionWriter.WriteByte(cie.ReturnAddressRegister < 0x7F ? (byte)1u : (byte)3u); // Version
+            _sectionWriter.Write(augmentationString.UnderlyingArray);
+
+            _sectionWriter.WriteULEB128(cie.CodeAlignFactor);
+            _sectionWriter.WriteSLEB128(cie.DataAlignFactor);
+            _sectionWriter.WriteULEB128(cie.ReturnAddressRegister);
+
+            _sectionWriter.WriteULEB128(augmentationLength);
+            if (cie.PersonalitySymbolName != null)
+            {
+                _sectionWriter.WriteByte(cie.PersonalityEncoding);
+                WriteAddress(cie.PersonalityEncoding, cie.PersonalitySymbolName);
+            }
+            if (cie.LsdaEncoding != 0)
+            {
+                _sectionWriter.WriteByte(cie.LsdaEncoding);
+            }
+            if (cie.PointerEncoding != 0)
+            {
+                _sectionWriter.WriteByte(cie.PointerEncoding);
+            }
+
+            _sectionWriter.Write(cie.Instructions);
+
+            _sectionWriter.Write(stackalloc byte[(int)padding]);
```

Comment:
is the stackalloc to ensure that padding is zero-initialized? I think adding `WriteZeroes` would be more explicit. Also will not have problems with  `SkipLocalsInitAttribute` if that gets applied somehow one day.

Reply:
I think I crossed the threshold of using the `.Write(stackalloc byte[...])` pattern more than 3 times, so a helper method is probably in order. Jan has previously wrote the same concern about zero initialization but I only fixed that one that one specific place.